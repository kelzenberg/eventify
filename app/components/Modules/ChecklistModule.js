import React from 'react';
import PropTypes from 'prop-types';

export default function ChecklistModule(props) {
    let moduleID = props.moduleData.type + "_"+props.moduleData.name;

    return <>
        <table className="table table-borderless table-hover align-middle">
            <thead>
                <tr>
                <th scope="col" style={{width: "15px"}}></th>
                <th scope="col">List Item</th>
                <th scope="col" className="text-end">Assignees</th>
                <th scope="col" style={{width: "15px"}}></th>
                </tr>
            </thead>
            <tbody>
                {props.moduleData.items.map(item => <tr key={item.name} aria-labelledby={moduleID + "_name"}>
                    <th scope="row">
                        <input className="form-check-input" type="checkbox" aria-label="Item Done"/>
                    </th>
                    <td>
                        <span style={{fontWeight: 500}} id={moduleID + "_name"}>{item.name}</span>
                        <br/>
                        <span>{item.description}</span>
                    </td>
                    <td className="text-end">
                        {item.assignees.map(userID => <div
                            key={userID}
                            role="button"
                            aria-label={"Unassign " + (props.event.members.find(m => m.id == userID).name)}
                            title={"Unassign " + (props.event.members.find(m => m.id == userID).name)}
                            className="iconBox iconBox-removable me-2"
                            style={{backgroundImage: "url(/assets/icons/user-image.svg)"}}
                        >
                            {/* <img src={`/api/users/${userID}/image`} alt=""/> */}
                        </div>)}
                        
                        <div
                            className="iconBox"
                            role="button"
                            title="Assign Attendee"
                            aria-label="Assign Attendee"
                            style={{backgroundImage: "url(/assets/icons/add.svg)"}}
                        />
                    </td>
                    <td>
                        <img className="p-2" role="button" title="Remove Item" aria-label="Remove Item" src="/assets/icons/remove.svg"/>
                    </td>
                </tr>)}
            </tbody>
        </table>
        <div className="rounded border-dashed border-gray w-100 p-3 text-center text-primary" role="button" aria-label="Add new Item" onClick={null}>
            <img src="/assets/icons/add.svg" alt="" className="pe-2" style={{verticalAlign: "sub"}}/>
            ADD NEW ITEM
        </div>
    </>
}

ChecklistModule.propTypes = {
    event: PropTypes.object,
    moduleData: PropTypes.object
}