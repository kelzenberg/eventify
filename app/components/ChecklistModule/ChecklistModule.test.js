import React from 'react';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import ChecklistModule from './ChecklistModule';

const event = {
    id: "123456789",
    title: "Event Title",
    startedAt: new Date("2021-02-27T23:38:43.000Z"),
    endedAt: new Date("2021-06-27T23:38:43.000Z"),
    amountOfUsers: 5,
    description: "Event Description",
    members: [{
        name: "Calliope",
        id: 0,
    }, {
        name: "Elliot",
        id: 1
    }, {
        name: "Heinz-Dieter",
        id: 2
    }],
};

const checklistModule = {
    type: "checklist",
    name: "Packing list",
    items: [
        {
            name: "Toothpaste",
            description: "The fresh-minty one!",
            done: false,
            assignees: [0, 2]
        }, {
            name: "Sunscreen",
            description: "I am burning already...",
            done: true,
            assignees: []
        }, {
            name: "Music Speaker",
            description: "Let's listen to Genesis",
            done: false,
            assignees: [1]
        }, {
            name: "Snorkel Equipment",
            description: "",
            done: false,
            assignees: []
        }
    ]
};

test('Snapshot', () => {
    let { container } = render(
        <ChecklistModule event={event} moduleData={checklistModule} htmlID="SomeID"/>
    );

    expect(container).toMatchSnapshot();
});