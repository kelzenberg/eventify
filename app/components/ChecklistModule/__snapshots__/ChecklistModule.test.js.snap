// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div>
  <table
    class="table table-borderless table-hover align-middle"
  >
    <thead>
      <tr>
        <th
          scope="col"
          style="width: 15px;"
        />
        <th
          scope="col"
        >
          List Item
        </th>
        <th
          class="text-end"
          scope="col"
        >
          Assignees
        </th>
        <th
          scope="col"
          style="width: 15px;"
        />
      </tr>
    </thead>
    <tbody>
      <tr
        aria-labelledby="SomeID_name"
      >
        <th
          scope="row"
        >
          <input
            aria-label="Item Done"
            class="form-check-input"
            type="checkbox"
          />
        </th>
        <td>
          <span
            id="SomeID_name"
            style="font-weight: 500;"
          >
            Toothpaste
          </span>
          <br />
          <span>
            The fresh-minty one!
          </span>
        </td>
        <td
          class="text-end"
        >
          <div
            aria-label="Unassign Calliope"
            class="iconBox iconBox-removable me-2"
            role="button"
            style="background-image: url(/assets/icons/user-image.svg);"
            title="Unassign Calliope"
          />
          <div
            aria-label="Unassign Heinz-Dieter"
            class="iconBox iconBox-removable me-2"
            role="button"
            style="background-image: url(/assets/icons/user-image.svg);"
            title="Unassign Heinz-Dieter"
          />
          <div
            aria-label="Assign Attendee"
            class="iconBox"
            role="button"
            style="background-image: url(/assets/icons/add.svg);"
            title="Assign Attendee"
          />
        </td>
        <td>
          <img
            aria-label="Remove Item"
            class="p-2"
            role="button"
            src="/assets/icons/remove.svg"
            title="Remove Item"
          />
        </td>
      </tr>
      <tr
        aria-labelledby="SomeID_name"
      >
        <th
          scope="row"
        >
          <input
            aria-label="Item Done"
            class="form-check-input"
            type="checkbox"
          />
        </th>
        <td>
          <span
            id="SomeID_name"
            style="font-weight: 500;"
          >
            Sunscreen
          </span>
          <br />
          <span>
            I am burning already...
          </span>
        </td>
        <td
          class="text-end"
        >
          <div
            aria-label="Assign Attendee"
            class="iconBox"
            role="button"
            style="background-image: url(/assets/icons/add.svg);"
            title="Assign Attendee"
          />
        </td>
        <td>
          <img
            aria-label="Remove Item"
            class="p-2"
            role="button"
            src="/assets/icons/remove.svg"
            title="Remove Item"
          />
        </td>
      </tr>
      <tr
        aria-labelledby="SomeID_name"
      >
        <th
          scope="row"
        >
          <input
            aria-label="Item Done"
            class="form-check-input"
            type="checkbox"
          />
        </th>
        <td>
          <span
            id="SomeID_name"
            style="font-weight: 500;"
          >
            Music Speaker
          </span>
          <br />
          <span>
            Let's listen to Genesis
          </span>
        </td>
        <td
          class="text-end"
        >
          <div
            aria-label="Unassign Elliot"
            class="iconBox iconBox-removable me-2"
            role="button"
            style="background-image: url(/assets/icons/user-image.svg);"
            title="Unassign Elliot"
          />
          <div
            aria-label="Assign Attendee"
            class="iconBox"
            role="button"
            style="background-image: url(/assets/icons/add.svg);"
            title="Assign Attendee"
          />
        </td>
        <td>
          <img
            aria-label="Remove Item"
            class="p-2"
            role="button"
            src="/assets/icons/remove.svg"
            title="Remove Item"
          />
        </td>
      </tr>
      <tr
        aria-labelledby="SomeID_name"
      >
        <th
          scope="row"
        >
          <input
            aria-label="Item Done"
            class="form-check-input"
            type="checkbox"
          />
        </th>
        <td>
          <span
            id="SomeID_name"
            style="font-weight: 500;"
          >
            Snorkel Equipment
          </span>
          <br />
          <span />
        </td>
        <td
          class="text-end"
        >
          <div
            aria-label="Assign Attendee"
            class="iconBox"
            role="button"
            style="background-image: url(/assets/icons/add.svg);"
            title="Assign Attendee"
          />
        </td>
        <td>
          <img
            aria-label="Remove Item"
            class="p-2"
            role="button"
            src="/assets/icons/remove.svg"
            title="Remove Item"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <div
    aria-label="Add new Item"
    class="rounded border-dashed border-gray w-100 p-3 text-center text-primary"
    role="button"
  >
    <img
      alt=""
      class="pe-2"
      src="/assets/icons/add.svg"
      style="vertical-align: sub;"
    />
    ADD NEW ITEM
  </div>
</div>
`;
